osu.Game/Graphics/OsuFont.cs

```diff
@@ -72,25 +75,17 @@ public static string GetFamilyString(Typeface typeface)
         /// <summary>
         /// Retrieves the string representation of a <see cref="FontWeight"/>.
         /// </summary>
-        /// <param name="typeface">The <see cref="Typeface"/>.</param>
-        /// <param name="weight">The <see cref="FontWeight"/>.</param>
-        /// <returns>The string representation of <paramref name="weight"/> in the specified <paramref name="typeface"/>.</returns>
-        public static string GetWeightString(Typeface typeface, FontWeight weight)
+        /// <param name="family">The font family.</param>
+        /// <param name="weight">The font weight.</param>
+        /// <returns>The string representation of <paramref name="weight"/> in the specified <paramref name="family"/>.</returns>
+        public static string GetWeightString(string family, FontWeight weight)
         {
-            if (typeface == Typeface.Torus && weight == FontWeight.Medium)
+            if ((family == @"Torus" || family == @"Torus-Alternate") && weight == FontWeight.Medium)
```

Comment:
What's the function of changing these from matching on `Typeface` to `string`?

Reply:
Having them as `Typeface` makes it hard to retrieve the weight string from a `With` call.

Reply:
Was also going to suggest this but it would require a call to `GetFamilyString` at the moment, which seems a bit ugly. Maybe the naming should be moved to `Description` or some other attribute.

Reply:
I'm so confused.

> would require a call to GetFamilyString at the moment

This is already being done, it was moved to `GetFont()` from inside here.

> Having them as Typeface makes it hard to retrieve the weight string from a With call.

I also don't understand how this is impacting anything.

All I'm asking is why the changes in this PR weren't just:
```diff
diff --git a/osu.Game/Graphics/OsuFont.cs b/osu.Game/Graphics/OsuFont.cs
index edb484021c..0c6dc5ba95 100644
--- a/osu.Game/Graphics/OsuFont.cs
+++ b/osu.Game/Graphics/OsuFont.cs
@@ -77,7 +77,7 @@ public static string GetFamilyString(Typeface typeface)
         /// <returns>The string representation of <paramref name="weight"/> in the specified <paramref name="typeface"/>.</returns>
         public static string GetWeightString(Typeface typeface, FontWeight weight)
         {
-            if (typeface == Typeface.Torus && weight == FontWeight.Medium)
+            if ((typeface == Typeface.Torus || typeface == Typeface.TorusAlternate) && weight == FontWeight.Medium)
                 // torus doesn't have a medium; fallback to regular.
                 weight = FontWeight.Regular;
```
What am I missing here? Either I'm missing something or the changes in this PR are nondescript.

Reply:
Have you tried applying in IDE? there's a bit more to the flow (via the `With` part) and how it falls back to a string rather than an enum..

Reply:
Right, I see now.

Reply:
Yeah I intentionally separated out all the other changes to a commit with a reason why, since it’s hard to understand with only diff.